File: Iteration-2/src/Student.java
import java.util.ArrayList;
import java.util.List;

public class Student extends Person {
    private String studentID;
    private String password;
    private double GPA;
    private int semester;
    List<Course> courses; 
    private Transcript transcript;


    public Student(String firstName, String lastName, String studentID, String password, double gPA, int semester) {
        super(firstName, lastName);
        this.studentID = studentID;
        this.password = password;
        this.GPA = gPA;
        this.semester = semester;
        this.courses = new ArrayList<>();
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public double getGPA() {
        return GPA;
    }

    public void setGPA(double GPA) {
        this.GPA = GPA;
    }

    public String getStudentID() {
        return studentID;
    }

    public void setStudentID(String studentID) {
        this.studentID = studentID;
    }

    public int getSemester() {
        return semester;
    }

    public void setSemester(int semester) {
        this.semester = semester;
    }

    public void addCourse(Course course) {
        courses.add(course);
    }

    public List<Course> getCourses() {
        return courses;
    }
    public Transcript getTranscript() {
        return transcript;
    }
    public void setTranscript(Transcript transcript) {
        this.transcript = transcript;
    }

    public String createEmailAddress() {
        return getFirstName().toLowerCase() + getLastName().toLowerCase() + "@marun.edu.tr" ;
    }
}

File: Iteration-3/Src/Student.py
from Person import Person


class Student(Person):
    def __init__(self, first_name, last_name, student_id, password, gpa, semester):
        super().__init__(first_name, last_name)
        self.student_id = student_id
        self.password = password
        self.gpa = gpa
        self.semester = semester
        self.courses = []
        self.transcript = None

    def get_password(self):
        return self.password

    def set_password(self, password):
        self.password = password

    def get_gpa(self):
        return self.gpa

    def set_gpa(self, gpa):
        self.gpa = gpa

    def get_student_id(self):
        return self.student_id

    def set_student_id(self, student_id):
        self.student_id = student_id

    def get_semester(self):
        return self.semester

    def set_semester(self, semester):
        self.semester = semester

    def add_course(self, course):
        self.courses.append(course)

    def get_courses(self):
        return self.courses

    def get_transcript(self):
        return self.transcript

    def get_first_name(self):
        return self.first_name

    def get_last_name(self):
        return self.last_name

    def set_transcript(self, transcript):
        self.transcript = transcript

    def create_email_address(self):
        return self.first_name.lower() + self.last_name.lower() + "@marun.edu.tr"
